Hints.py

```diff
@@ -1024,10 +1060,41 @@ def buildWorldGossipHints(spoiler, world, checkedLocations=None):
 
     hint_types, hint_prob = zip(*hint_dist.items())
     hint_prob, _ = zip(*hint_prob)
+    
+    # Add required dual location hints, only if hint copies > 0
+    if hint_dist['dual_always'][1] > 0:
```

Comment:
This breaks with hint type order of 0 because the hint distro is filtered to non-zero orders when sorting. 
```suggestion
    if 'dual_always' in hint_dist and hint_dist['dual_always'][1] > 0:
```